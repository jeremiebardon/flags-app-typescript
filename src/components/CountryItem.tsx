import { AspectRatio, Box, Image, List, ListItem, Skeleton, Stack, Text } from "@chakra-ui/react"
import { formatNumberWithComma } from "../helpers/math";
import { Country } from "../types/countries.type";

import { Link } from "react-router-dom";

interface CountryItemProps {
  country: Country;
}

export const CountryItem = ({ country }: CountryItemProps) => {
  return (
    <Link to={`/country/${country.cca3}`}>
      <Stack bg="bg-surface" height='100%' shadow='sm'>
        <Box position="relative">
          <AspectRatio ratio={4 / 3}>
            <Image
              src={country.flags.png}
              alt={`${country.name.common}`}
              draggable="false"
              fallback={<Skeleton />}
            />
          </AspectRatio>
        </Box>
        <Stack justifyContent='space-between' display='flex' flexDirection='column' p={4} mt={0} height='100%'>
          <Text fontSize='md' fontWeight='600' mb={2}> { country.name.common } </Text>
          <List>
            <ListItem> 
              <Text fontSize='sm'> <b> Population: </b> {formatNumberWithComma(country.population)} </Text>
            </ListItem>
            <ListItem>
              <Text fontSize='sm'> <b> Region: </b> { country.region } </Text>
            </ListItem>
            <ListItem>
              <Text fontSize='sm'> <b> Capital: </b> { country.capital ?? 'Unknown' } </Text>
            </ListItem>
          </List>
        </Stack>
      </Stack>
    </Link>
  )
}