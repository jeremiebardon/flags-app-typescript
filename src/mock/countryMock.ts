import { BorderCountry, Country } from "../types/countries.type";

export const countryMockFrance = {
  name: {
    common: "France",
    official: "French Republic",
    nativeName: {
      fra: {
        official: "R√©publique fran√ßaise",
        common: "France",
      },
    },
  },
  tld: [".fr"],
  cca2: "FR",
  ccn3: "250",
  cca3: "FRA",
  cioc: "FRA",
  independent: true,
  status: "officially-assigned",
  unMember: true,
  currencies: {
    EUR: {
      name: "Euro",
      symbol: "‚Ç¨",
    },
  },
  idd: {
    root: "+3",
    suffixes: ["3"],
  },
  capital: ["Paris"],
  altSpellings: ["FR", "French Republic", "R√©publique fran√ßaise"],
  region: "Europe",
  subregion: "Western Europe",
  languages: {
    fra: "French",
  },
  translations: {
    ara: {
      official: "ÿßŸÑÿ¨ŸÖŸáŸàÿ±Ÿäÿ© ÿßŸÑŸÅÿ±ŸÜÿ≥Ÿäÿ©",
      common: "ŸÅÿ±ŸÜÿ≥ÿß",
    },
    ces: {
      official: "Francouzsk√° republika",
      common: "Francie",
    },
    cym: {
      official: "French Republic",
      common: "France",
    },
    deu: {
      official: "Franz√∂sische Republik",
      common: "Frankreich",
    },
    est: {
      official: "Prantsuse Vabariik",
      common: "Prantsusmaa",
    },
    fin: {
      official: "Ranskan tasavalta",
      common: "Ranska",
    },
    fra: {
      official: "R√©publique fran√ßaise",
      common: "France",
    },
    hrv: {
      official: "Francuska Republika",
      common: "Francuska",
    },
    hun: {
      official: "Francia K√∂zt√°rsas√°g",
      common: "Franciaorsz√°g",
    },
    ita: {
      official: "Repubblica francese",
      common: "Francia",
    },
    jpn: {
      official: "„Éï„É©„É≥„ÇπÂÖ±ÂíåÂõΩ",
      common: "„Éï„É©„É≥„Çπ",
    },
    kor: {
      official: "ÌîÑÎûëÏä§ Í≥µÌôîÍµ≠",
      common: "ÌîÑÎûëÏä§",
    },
    nld: {
      official: "Franse Republiek",
      common: "Frankrijk",
    },
    per: {
      official: "ÿ¨ŸÖŸáŸàÿ±€å ŸÅÿ±ÿßŸÜÿ≥Ÿá",
      common: "ŸÅÿ±ÿßŸÜÿ≥Ÿá",
    },
    pol: {
      official: "Republika Francuska",
      common: "Francja",
    },
    por: {
      official: "Rep√∫blica Francesa",
      common: "Fran√ßa",
    },
    rus: {
      official: "–§—Ä–∞–Ω—Ü—É–∑—Å–∫–∞—è –†–µ—Å–ø—É–±–ª–∏–∫–∞",
      common: "–§—Ä–∞–Ω—Ü–∏—è",
    },
    slk: {
      official: "Franc√∫zska republika",
      common: "Franc√∫zsko",
    },
    spa: {
      official: "Rep√∫blica franc√©s",
      common: "Francia",
    },
    swe: {
      official: "Republiken Frankrike",
      common: "Frankrike",
    },
    urd: {
      official: "ÿ¨ŸÖ€ÅŸàÿ±€å€Å ŸÅÿ±ÿßŸÜÿ≥",
      common: "ŸÅÿ±ÿßŸÜÿ≥",
    },
    zho: {
      official: "Ê≥ïÂÖ∞Ë•øÂÖ±ÂíåÂõΩ",
      common: "Ê≥ïÂõΩ",
    },
  },
  latlng: [46, 2],
  landlocked: false,
  borders: ["AND", "BEL", "DEU", "ITA", "LUX", "MCO", "ESP", "CHE"],
  area: 551695,
  demonyms: {
    eng: {
      f: "French",
      m: "French",
    },
    fra: {
      f: "Fran√ßaise",
      m: "Fran√ßais",
    },
  },
  flag: "üá´üá∑",
  maps: {
    googleMaps: "https://goo.gl/maps/g7QxxSFsWyTPKuzd7",
    openStreetMaps: "https://www.openstreetmap.org/relation/1403916",
  },
  population: 67391582,
  gini: {
    "2018": 32.4,
  },
  fifa: "FRA",
  car: {
    signs: ["F"],
    side: "right",
  },
  timezones: [
    "UTC-10:00",
    "UTC-09:30",
    "UTC-09:00",
    "UTC-08:00",
    "UTC-04:00",
    "UTC-03:00",
    "UTC+01:00",
    "UTC+02:00",
    "UTC+03:00",
    "UTC+04:00",
    "UTC+05:00",
    "UTC+10:00",
    "UTC+11:00",
    "UTC+12:00",
  ],
  continents: ["Europe"],
  flags: {
    png: "https://flagcdn.com/w320/fr.png",
    svg: "https://flagcdn.com/fr.svg",
  },
  coatOfArms: {
    png: "https://mainfacts.com/media/images/coats_of_arms/fr.png",
    svg: "https://mainfacts.com/media/images/coats_of_arms/fr.svg",
  },
  startOfWeek: "monday",
  capitalInfo: {
    latlng: [48.87, 2.33],
  },
  postalCode: {
    format: "#####",
    regex: "^(\\d{5})$",
  },
} as Country;

export const borderCountries = [
  {
    name: {
      common: "Andorra",
      official: "Principality of Andorra",
      nativeName: {
        cat: {
          official: "Principat d'Andorra",
          common: "Andorra",
        },
      },
    },
    cca3: "AND",
  },
  {
    name: {
      common: "Spain",
      official: "Kingdom of Spain",
      nativeName: {
        spa: {
          official: "Reino de Espa√±a",
          common: "Espa√±a",
        },
      },
    },
    cca3: "ESP",
  },
  {
    name: {
      common: "Switzerland",
      official: "Swiss Confederation",
      nativeName: {
        fra: {
          official: "Conf√©d√©ration suisse",
          common: "Suisse",
        },
        gsw: {
          official: "Schweizerische Eidgenossenschaft",
          common: "Schweiz",
        },
        ita: {
          official: "Confederazione Svizzera",
          common: "Svizzera",
        },
        roh: {
          official: "Confederaziun svizra",
          common: "Svizra",
        },
      },
    },
    cca3: "CHE",
  },
  {
    name: {
      common: "Italy",
      official: "Italian Republic",
      nativeName: {
        ita: {
          official: "Repubblica italiana",
          common: "Italia",
        },
      },
    },
    cca3: "ITA",
  },
  {
    name: {
      common: "Monaco",
      official: "Principality of Monaco",
      nativeName: {
        fra: {
          official: "Principaut√© de Monaco",
          common: "Monaco",
        },
      },
    },
    cca3: "MCO",
  },
  {
    name: {
      common: "Germany",
      official: "Federal Republic of Germany",
      nativeName: {
        deu: {
          official: "Bundesrepublik Deutschland",
          common: "Deutschland",
        },
      },
    },
    cca3: "DEU",
  },
  {
    name: {
      common: "Belgium",
      official: "Kingdom of Belgium",
      nativeName: {
        deu: {
          official: "K√∂nigreich Belgien",
          common: "Belgien",
        },
        fra: {
          official: "Royaume de Belgique",
          common: "Belgique",
        },
        nld: {
          official: "Koninkrijk Belgi√´",
          common: "Belgi√´",
        },
      },
    },
    cca3: "BEL",
  },
  {
    name: {
      common: "Luxembourg",
      official: "Grand Duchy of Luxembourg",
      nativeName: {
        deu: {
          official: "Gro√üherzogtum Luxemburg",
          common: "Luxemburg",
        },
        fra: {
          official: "Grand-Duch√© de Luxembourg",
          common: "Luxembourg",
        },
        ltz: {
          official: "Groussherzogtum L√´tzebuerg",
          common: "L√´tzebuerg",
        },
      },
    },
    cca3: "LUX",
  },
] as BorderCountry[];
