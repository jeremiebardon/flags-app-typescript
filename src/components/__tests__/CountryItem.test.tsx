import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { countryMockFrance } from "../../mock/countryMock";
import { CountryItem } from "../CountryItem";

describe('<CountryItem />', () => {
  it('should render', () => {
    const wrapper = render(<CountryItem country={countryMockFrance} />, {
      wrapper: BrowserRouter
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render without region', () => {
    const { getByText } = render(<CountryItem country={{ ...countryMockFrance, capital: undefined }} />, {
      wrapper: BrowserRouter
    });

    const capital = getByText('Unknown')

    expect(capital).toBeInTheDocument();
  });
});
