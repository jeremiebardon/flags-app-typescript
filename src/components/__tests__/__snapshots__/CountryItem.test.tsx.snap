// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryItem /> should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/country/FRA"
      >
        <div
          class="chakra-stack css-1527vj0"
        >
          <div
            class="css-79elbk"
          >
            <div
              class="chakra-aspect-ratio css-791950"
            >
              <div
                class="chakra-skeleton css-1twknw3"
              />
            </div>
          </div>
          <div
            class="chakra-stack css-woqlpa"
          >
            <p
              class="chakra-text css-je8k2y"
            >
               
              France
               
            </p>
            <ul
              class="css-155za0w"
              role="list"
            >
              <li
                class="css-0"
              >
                <p
                  class="chakra-text css-g5hezf"
                >
                   
                  <b>
                     Population: 
                  </b>
                   
                  67,391,582
                   
                </p>
              </li>
              <li
                class="css-0"
              >
                <p
                  class="chakra-text css-g5hezf"
                >
                   
                  <b>
                     Region: 
                  </b>
                   
                  Europe
                   
                </p>
              </li>
              <li
                class="css-0"
              >
                <p
                  class="chakra-text css-g5hezf"
                >
                   
                  <b>
                     Capital: 
                  </b>
                   
                  Paris
                   
                </p>
              </li>
            </ul>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/country/FRA"
    >
      <div
        class="chakra-stack css-1527vj0"
      >
        <div
          class="css-79elbk"
        >
          <div
            class="chakra-aspect-ratio css-791950"
          >
            <div
              class="chakra-skeleton css-1twknw3"
            />
          </div>
        </div>
        <div
          class="chakra-stack css-woqlpa"
        >
          <p
            class="chakra-text css-je8k2y"
          >
             
            France
             
          </p>
          <ul
            class="css-155za0w"
            role="list"
          >
            <li
              class="css-0"
            >
              <p
                class="chakra-text css-g5hezf"
              >
                 
                <b>
                   Population: 
                </b>
                 
                67,391,582
                 
              </p>
            </li>
            <li
              class="css-0"
            >
              <p
                class="chakra-text css-g5hezf"
              >
                 
                <b>
                   Region: 
                </b>
                 
                Europe
                 
              </p>
            </li>
            <li
              class="css-0"
            >
              <p
                class="chakra-text css-g5hezf"
              >
                 
                <b>
                   Capital: 
                </b>
                 
                Paris
                 
              </p>
            </li>
          </ul>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
